listenbrainz/webserver/static/js/src/PinRecordingModal.tsx

```diff
@@ -0,0 +1,174 @@
+    super(props);
+    this.state = { blurbContent: undefined };
+  }
+
+  submitPinRecording = async () => {
+    const { recordingToPin, isCurrentUser, newAlert } = this.props;
+    const { blurbContent } = this.state;
+    const { APIService, currentUser } = this.context;
+
+    if (isCurrentUser && currentUser?.auth_token) {
+      const recordingMSID = _get(
+        recordingToPin,
+        "track_metadata.additional_info.recording_msid"
+      );
+      const recordingMBID = _get(
+        recordingToPin,
+        "track_metadata.additional_info.recording_mbid"
+      );
+
+      try {
+        const status = await APIService.submitPinRecording(
+          currentUser.auth_token,
+          recordingMSID,
+          recordingMBID || undefined,
+          blurbContent || undefined
+        );
+        if (status === 200) {
+          newAlert(
+            "success",
+            `You pinned a recording!`,
+            `${recordingToPin?.track_metadata.artist_name} - ${recordingToPin?.track_metadata.track_name}`
+          );
+          this.setState({ blurbContent: undefined });
+        }
+      } catch (error) {
+        this.handleError(error, "Error while pinning recording");
+      }
+    }
+  };
+
+  handleBlurbInputChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    event.preventDefault();
+    const input = event.target.value.replace(/\s\s+/g, " "); // remove line breaks and excessive spaces
+    if (input.length <= this.maxBlurbContentLength) {
+      this.setState({ blurbContent: input });
+    }
+  };
+
+  handleError = (error: string | Error, title?: string): void => {
+    const { newAlert } = this.props;
+    if (!error) {
+      return;
+    }
+    newAlert(
+      "danger",
+      title || "Error",
+      typeof error === "object" ? error.message : error
+    );
+  };
+
+  render() {
+    const { recordingToPin } = this.props;
+    const { blurbContent } = this.state;
+    const track_name = recordingToPin?.track_metadata?.track_name;
+    const artist_name = recordingToPin?.track_metadata?.artist_name;
+
+    return (
+      <div
+        className="modal fade"
+        id="PinRecordingModal"
+        tabIndex={-1}
+        role="dialog"
+        aria-labelledby="PinRecordingModalLabel"
+        data-backdrop="static"
+      >
+        <div className="modal-dialog" role="document">
+          <form className="modal-content">
+            <div className="modal-header">
+              <button
+                type="button"
+                className="close"
+                data-dismiss="modal"
+                aria-label="Close"
+              >
+                <span aria-hidden="true">&times;</span>
+              </button>
+              <h4 className="modal-title" id="PinRecordingModalLabel">
+                Pin This Recording to Your Profile
+              </h4>
+            </div>
+            <div className="modal-body">
+              <p>
+                Why are you pinning{" "}
```

Comment:
(nit)

This sentence reads a bit weird to me, it can also be taken in a negative sense. Could we make the sentence more obviously positive?

"Why do you love this song?" or something similar? I also notice that the placeholder is along these lines, so perhaps we're okay with this.

Reply:
Changed the message to make the purpose of pinning recordings a bit clearer.